sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAppRoleStatelessAuthenticationFilter.java

```diff
@@ -65,10 +67,14 @@ protected void doFilterInternal(HttpServletRequest httpServletRequest,
         }
         try {
             final UserPrincipal userPrincipal = principalManager.buildUserPrincipal(aadIssuedBearerToken);
+            Set<SimpleGrantedAuthority> authorities = toSimpleGrantedAuthoritySet(userPrincipal);
+            if(userPrincipal.getClaim(AADTokenClaim.SCP) != null){
```

Comment:
Should we move this condition into the scopeToSimpleGrantedAuthoritySet?

Reply:
The `authorities` returned by the first function could be a unmodifiableSet. If we try to add scopes here, an exception will be thrown.

Reply:
Maybe we should combine these two functions together.

Reply:
The combine will look even better.